chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -95,14 +105,50 @@ def __exit__(self, type, value, traceback):
             self.unlock_pool()
 
     def lock_pool(self):
-        lock = self.import_locks[self.pool_path]
-        lock.acquire()
-        # log.debug('ZfsDevice acquired lock on %s' % self.pool_path)
+        """
+        Attempt to acquire a lock on a given zpool through a lockfile, increment the reference count each time
+        this method is called, regardless of whether acquire is actually called. Keep trying until we know
+        we are locking the zpool.
+
+        Lock acquire polls the timeout at intervals of LOCK_ACQUIRE_TIMEOUT.
+        """
+        if ZfsDevice.locks_dir_initialized == False:
+            # ensure lock directory exists
+            try:
+                os.mkdir(self.ZPOOL_LOCK_DIR)
+            except OSError:
+                pass
+
+            ZfsDevice.locks_dir_initialized = True
+
+        while not self.lock.i_am_locking():
```

Comment:
logging has not been added to ZfsDevice in this patch on purpose to keep the patch to minimum changes necessary. I have various other versions with extra logging added and that can be added as a follow-on patch. regarding your other point, I've considered adding timeout (and previous iterations did have one), but that should never happen (lock forever) and so timeout is unnecessary.